src/cpu/aarch64/shuffle/jit_uni_shuffle.cpp

```diff
@@ -58,7 +58,10 @@ status_t jit_uni_shuffle_t<isa>::pd_t::init(engine_t *engine) {
     if (blocked_format == format_tag::undef) return status::unimplemented;
 
     conf_.blk_size = src_d.blocking_desc().strides[ndims() - 1];
-    conf_.simd_w = cpu_isa_traits<isa>::vlen / sizeof(float);
+    /* Because "ST1H { <Zt>.S }, <Pg>, [<Xn|SP>, <Zm>.S, <mod> #1]" is used
+       to gather data for bf16, simd_w must be calculated
+       with sizeof(unsigned). */
+    conf_.simd_w = cpu_isa_traits<isa>::vlen / sizeof(unsigned);
```

Comment:
I'm not sure I follow this change. According to the [standard ](https://en.cppreference.com/w/cpp/language/types) this type has variadic length depending on a system. If this is intended to be `2`, I'd suggest to use `bfloat16_t` type directly here.

Reply:
you can also use `data_type_size(bf16)` from type_helpers.hpp.

Reply:
Thank you for the comment. Here, I intended to be 4 for both of fp32 and bf16, because the data gather instructions for fp32 and bf16 can have 32-bit vectored address offset in AArch64 ISA.
I changed `sizeof(unsigned)` -> `data_type_size(s32)`.